expfmt/openmetrics_parse_test.go

```diff
@@ -0,0 +1,307 @@
+					Unit: proto.String("seconds"),
+					Help: proto.String("abc"),
+				},
+			},
+		},
+
+		// 4: type gauge
+		{
+			in: `# HELP foo_seconds abc
+# TYPE foo_seconds gauge
+# UNIT foo_seconds seconds
+# EOF
+`,
+
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("foo_seconds"),
+					Type: dto.MetricType_GAUGE.Enum(),
+					Unit: proto.String("seconds"),
+					Help: proto.String("abc"),
+				},
+			},
+		},
+
+		// 5: type histogram
+		{
+			in: `# HELP foo_seconds abc
+# TYPE foo_seconds histogram
+# UNIT foo_seconds seconds
+# EOF
+`,
+
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("foo_seconds"),
+					Type: dto.MetricType_HISTOGRAM.Enum(),
+					Unit: proto.String("seconds"),
+					Help: proto.String("abc"),
+				},
+			},
+		},
+
+		// 6: type summary
+		{
+			in: `# HELP foo_seconds abc
+# TYPE foo_seconds summary
+# UNIT foo_seconds seconds
+# EOF
+`,
+
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("foo_seconds"),
+					Type: dto.MetricType_SUMMARY.Enum(),
+					Unit: proto.String("seconds"),
+					Help: proto.String("abc"),
+				},
+			},
+		},
+
+		// 7: type summary
+		{
+			in: `# HELP foo_seconds abc
+# TYPE foo_seconds gaugehistogram
+# UNIT foo_seconds seconds
+# EOF
+`,
+
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("foo_seconds"),
+					Type: dto.MetricType_GAUGE_HISTOGRAM.Enum(),
+					Unit: proto.String("seconds"),
+					Help: proto.String("abc"),
+				},
+			},
+		},
+
+		// 8: a normal comment
+		{
+			in: `#
+# TYPE name_seconds counter
+# UNIT name_seconds seconds
+# HELP name_seconds two-line\n doc  str\\ing
+# HELP  name2 doc str"ing 2
+#    TYPE    name2 gauge
+# EOF
+`,
+
+			out: []*dto.MetricFamily{
+				{
+					Name: proto.String("name_seconds"),
+					Type: dto.MetricType_COUNTER.Enum(),
+					Unit: proto.String("seconds"),
+					Help: proto.String("two-line\n doc  str\\ing"),
+				},
+				{
+					Name: proto.String("name2"),
+					Type: dto.MetricType_GAUGE.Enum(),
+					Help: proto.String("doc str\"ing 2"),
+				},
+			},
+		},
+	}
+
+	for i, scenario := range scenarios {
+		out, err := omParser.OpenMetricsToMetricFamilies(strings.NewReader(scenario.in))
+		if err != nil {
+			t.Errorf("%d. error: %s", i, err)
+			continue
+		}
+		if expected, got := len(scenario.out), len(out); expected != got {
+			t.Errorf(
+				"%d. expected %d MetricFamilies, got %d",
+				i, expected, got,
+			)
+		}
+		for _, expected := range scenario.out {
+			got, ok := out[expected.GetName()]
+			if !ok {
+				t.Errorf(
+					"%d. expected MetricFamily %q, found none",
+					i, expected.GetName(),
+				)
+				continue
+			}
+			if expected.String() != got.String() {
+				t.Errorf(
+					"%d. expected MetricFamily %s, got %s",
+					i, expected, got,
+				)
+			}
+		}
+	}
+}
+
+func testOpenMetricParseError(t testing.TB) {
+	scenarios := []struct {
+		in  string
+		err string
+	}{
+		// 0:
+		{
+			in: `# TYPE metric counter
+# TYPE metric untyped
+`,
```

Comment:
Let's do the same here! Adding `name` to the scenarios and then using t.Run should provide more context about what each scenario is about :)

Reply:
has a little confusing about this, because now there's benchmark for the test, so the arg in test function is testing.TB, and arg required by t.Run is *testing.T(`t.Run(name string, f func(t *testing.T))`), the testing.TB has not  implement the interface